src/main/java/com/pkb/unit/State.java

```diff
@@ -1,9 +1,35 @@
 package com.pkb.unit;
 
+/**
+ * Possible status a unit can be in.
+ */
 public enum State {
+    /**
+     * Failed state indicates that something went wrong
+     * with the unit so it is not operational but can be started.
+     */
     FAILED,
+
+    /**
+     * Stopped state indicates that the unit is not working but
+     * can be started.
+     */
     STOPPED,
+
+    /**
+     * Stopping state indicates that the unit is about to stop.
+     */
     STOPPING,
+
+    /**
+     * Started state indicates that the unit is fully operational and
+     * can be stopped.
+     */
     STARTED,
+
+    /**
+     * Starting state indicates that the unit is about to start.
+     * Usually it is waiting for the units to be started it depends on.
```

Comment:
This sounds strange.
Do you mean "to be started" as to be in the `STARTED` state or to have started?
Maybe something like:
```suggestion
     * Usually it is waiting for the units it depends on to have started.
```

Reply:
You are right, "to have started" is better. Thanks